shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/data/remote/datasource/SurveyRemoteDataSource.kt

```diff
@@ -24,4 +25,13 @@ class SurveyRemoteDataSourceImpl(private val apiClient: ApiClient) : SurveyRemot
             }
         )
     }
+
+    override fun getSurvey(id: String): Flow<SurveyApiModel> {
+        return apiClient.body(
+            HttpRequestBuilder().apply {
+                path("/v1/surveys/${id}")
+                method = HttpMethod.Get
+            }
```

Comment:
🤔 I think you want to reuse the `SurveyApiModel` but this is the request for fetching all survey detail, so I think we should create a new `ApiModel` to handle the response with `questions`. Ex: `SurveyDetailApiModel`, it's more neat and clean imo.

Reply:
@suho Hmm...We should have two different models in the domain as well, I know we are working on survey api and we already know which fields we have to use; with the bigger project, it would be cumbersome when the `Survey` model has a few properties and `SurveyDetail` provide a bunch of detailed information if we reuse the same Survey model.

Reply:
With this project, I am still fine if you want to combine it to a Survey model for reusing purposes. I just want to make it clear about your concern 😄

Reply:
Yes, with this project, I prefer to keep reusing `SurveyApiModel` in the data layer and `Survey` in the domain layer in this case. It uses the same business model (the same table from the database), so it will be easy for us to keep only one type. For example, I can update survey data with questions to realm when I fetch the survey detail.

Reply:
@suho I think for the `questions`, it would be in the new table since it's a different DTO and in the Survey we just need the ids for referencing.

> Yes, with this project, I prefer to keep reusing SurveyApiModel in the data layer and Survey in the domain layer in this case.

OK. Deal 🤝

Reply:
Thanks @hoangnguyen92dn 🫡